rust/src/memory.rs

```diff
@@ -68,6 +68,19 @@ pub fn free_aligned(p: *const u8) {
     }
 }
 
+pub fn memcpy(dst: *const u8, src: *const u8, len: usize) {
```

Comment:
Shouldn't `dst` be `*u8` rather than `*const u8`?

Reply:
I just extracted this from the original code. In fact I think we should change `allocate_aligned` to return `*mut u8` instead of `*const u8`, which will then allow us to change `*const` here.